docs/tech-specs/pull-consumer-spec.md

```diff
@@ -0,0 +1,42 @@
+# Tech Spec: Adding Pull Consumer Support
+
+<!-- |  |  |
+| -- | -- |
+| *Version* | 1 |
+| *By* | Imran M Yousuf |
+| *Date* | November 10, 2020 |
+| *Last Update* | December 30, 2020 | -->
+
+## Problem Statement / Goal
+
+In some use-cases pull based consumers are more desirable than push based ones. We want to give each consumer an option to register itself as push based or pull based.
+
+## Use cases
+
+In most cases push based consumers are preferable. But it might not be ideal for events that require long processing. We might handle long running events in the push consumers in two ways:
+
+- Process the event and then reply to the broker. For long running events, the broker will timeout before the processing completes.
+- Reply success to the webhook-broker immediately and process the event in background. If we do so, the webhook-broker might overwhelm the consumer with events.
+
+We can solve this problem with pull based consumers.
+
+### High-level Flow
+
+1. Each consumer will be classified as either **Push** or **Pull** based on a boolean flag _isPullConsumer_. Default value of the flag should be _False_ i.e: the consumers by default will be **Push** based.
```

Comment:
In Go, for the method to be usable externally the first alphabet must be capitalized. For example, `IsPullConsumer`